web/react/stores/post_store.jsx

```diff
@@ -99,8 +99,57 @@ var PostStore = assign({}, EventEmitter.prototype, {
         }
         return null;
     },
-    storePosts: function storePosts(channelId, posts) {
-        this.pStorePosts(channelId, posts);
+    storePosts: function storePosts(channelId, newPostList) {
+        if (isPostListNull(newPostList)) {
+            return;
+        }
+
+        var postList = PostStore.getPosts(channelId);
+        if (isPostListNull(postList)) {
+            postList = newPostList;
+        } else {
+            postList = makePostListNonNull(postList);
```

Comment:
Haha yeah, the real reason I did this was to skip the merge code (which was at first more complicated) if it's a totally new post list. I'm going to change it to be more reasonable